orchestrator/src/main/java/com/yahoo/vespa/orchestrator/policy/HostedVespaClusterPolicy.java

```diff
@@ -62,32 +81,110 @@ public void verifyGroupGoingDownPermanentlyIsFine(ClusterApi clusterApi)
+            //    7  admin                slobrok                       not checked  1-3 in jdisc container cluster
+            //    8  metrics              metricsproxy-container        Slobrok      application
+            //    9  admin                logd                          not checked  application
+            //   10  admin                config-sentinel               not checked  application
+            //   11  admin                configproxy                   not checked  application
+            //   12  admin                logforwarder                  not checked  application
+            //   13  controller           controller                    state/v1     controllers
+            //   14  zone-config-servers  configserver                  state/v1     config servers
+            //   15  controller-host      hostadmin                     state/v1     controller hosts
+            //   16  configserver-host    hostadmin                     state/v1     config server hosts
+            //   17  tenant-host          hostadmin                     state/v1     tenant hosts
+            //   18  proxy-host           hostadmin                     state/v1     proxy hosts
+            //
+            // CCN refers to the content cluster's name, as specified in services.xml.
+            // JCCN refers to the jdisc container cluster's name, as specified in services.xml.
+            //
+            // For instance a content node will have 2-5 and 8-12 and possibly 1, while a combined
+            // cluster node may have all 1-12.
+            //
+            // The services on a node can be categorized into these main types, ref association column above:
+            //   A  content
+            //   B  container
+            //   C  tenant host
+            //   D  config server
+            //   E  config server host
+            //   F  controller
+            //   G  controller host
+            //   H  proxy (same as B)
+            //   I  proxy host
+
+            if (clusterApi.serviceType().equals(ServiceType.CLUSTER_CONTROLLER)) {
+                // All nodes have all state and we need to be able to remove the half that are retired on cluster migration
+                return ConcurrentSuspensionLimitForCluster.FIFTY_PERCENT;
+            }
 
-        if (VespaModelUtil.CLUSTER_CONTROLLER_SERVICE_TYPE.equals(clusterApi.serviceType())) {
-            // All nodes have all state and we need to be able to remove the half that are retired on cluster migration
-            return ConcurrentSuspensionLimitForCluster.FIFTY_PERCENT;
-        }
+            if (Set.of(ServiceType.STORAGE, ServiceType.SEARCH, ServiceType.DISTRIBUTOR, ServiceType.TRANSACTION_LOG_SERVER)
+                    .contains(clusterApi.serviceType())) {
+                // Delegate to the cluster controller
+                return ConcurrentSuspensionLimitForCluster.ALL_NODES;
```

Comment:
Any estimates on how much increased load the cluster controller(s) will be getting after this change?

Reply:
Worst-case, all Docker hosts wants to reboot, and in case each content node may end up sending 2 requests (one probe and a follow-up non-probe) per 30s host-admin tick.  For large applications there are, say, 50 nodes, so QPS is 3.3.  This QPS would fairly flat, i.e. not bursty.  Once all hosts wants to reboot we'd hit this max QPS, and it would drop linearly as hosts were rebooted.

For a more realistic and common scenario, whenever a large application upgrades, they'll send 1 request/tick, i.e. a max QPS of 1.67 that will drop off linearly.

These high QPS MAY also appear today.  I looked at an application with 50 content nodes being upgraded:
 * The first request set node 92 to maintenance, but took longer than the 10s timeout to complete, so that node retried 30s later on the next tick, and when it got permission.
 * In the 20s window in between, there were 30s of accumulated queries: different nodes trying to get permission to suspend.  I counted 49 requests, i.e. QPS of 1.63 and confirming the theory.
 * However in the 10-13s window there was a burst of queries accumulated from 0-13s, and therefore 4x of normal high QPS.
 * The latencies were >10s for the queries that actually induce cluster state changes, which is expected.  All others have sub-0.1s response times, and spent cumulative 2.6s, which means the CC can handle sustained 20 QPS, or a cluster size of 600 for upgrades, or 300 for host suspensions.